cmd/rekor_monitor/main.go

```diff
@@ -82,6 +123,27 @@ func main() {
 		config.OutputIdentitiesFile = outputIdentitiesFileName
 	}
 
+	if runAsMonitor != nil && *runAsMonitor {
+		go func() {
```

Comment:
The main script has gotten a bit unwieldy unfortunately. Can we split the initialization of metrics and starting up the server into its own file, also so that we can add tests?

https://github.com/sigstore/rekor-tiles/blob/main/pkg/server/metrics.go is a decent example, just ignoring the bits about gRPC.

This also gives an example of gracefully shutting down the metrics server when sent a SIGINT/SIGTERM in case this is running on K8s.

It'd also be nice to not use promauto and initialize a new registry, in case this gets refactored into a library to avoid any collisions of prom registries.